src/utils/getTokenList.ts

```diff
@@ -89,6 +90,40 @@ export default async function getTokenList(
   throw new Error('Unrecognized list URL protocol.')
 }
 
+// loop to fetch all whitelist token
+export async function getTokenListV2(
+  listUrl: string,
+  resolveENSContentHash?: (ensName: string) => Promise<string>,
+): Promise<TokenList> {
+  return new Promise(async (resolve, reject) => {
+    let tokens: any[] = []
+    try {
+      const pageSize = 100
+      const maximumPage = 15
+      let page = 1
+      while (true) {
+        const { data } = await axios.get(`${listUrl}&pageSize=${pageSize}&page=${page}`)
+        page++
+        const tokensResponse = data.data.tokens ?? []
+        tokens = tokens.concat(tokensResponse)
+        if (tokensResponse.length < pageSize || page >= maximumPage) break // out of tokens, and prevent infinity loop
```

Comment:
User can scroll maximum 1500 tokens?

Reply:
this is fetching whitelist token, currently it is  < 100 tokens per chain.